hibernate-core/src/main/java/org/hibernate/engine/internal/Cascade.java

```diff
@@ -375,8 +388,35 @@ private static boolean isLogicalOneToOne(Type type) {
 		return type.isEntityType() && ( (EntityType) type ).isLogicalOneToOne();
 	}
 
-	private static boolean cascadeAssociationNow(final CascadePoint cascadePoint, AssociationType associationType) {
-		return associationType.getForeignKeyDirection().cascadeNow( cascadePoint );
+	private static boolean cascadeAssociationNow(
+			CascadingAction<?> action,
+			CascadePoint cascadePoint,
+			AssociationType associationType,
+			AttributeMapping attributeMapping,
+			boolean isStrictUnownedTransienceEnabled) {
+		return associationType.getForeignKeyDirection().cascadeNow( cascadePoint )
+				// For check on flush, we should only check owned associations when strictness is enforced
+				&& ( action != CHECK_ON_FLUSH || isStrictUnownedTransienceEnabled || !isUnownedAssociation( associationType, attributeMapping ) );
```

Comment:
It looks to me naively like this is going to disable the check even when the JPA bootstrap is used. I don't think we want to do that. This check has always occurred when using the JPA bootstrap, and is very explicitly required by the spec.

Reply:
Perfect, thanks.